trie/hexary.py

```diff
@@ -112,10 +112,16 @@ def __init__(self, db, root_hash=BLANK_NODE_HASH, prune=False):
         validate_is_bytes(root_hash)
         self.root_hash = root_hash
         self.is_pruning = prune
-        if prune:
-            self._ref_count = defaultdict(int)
+        if ref_count is None:
+            if prune:
+                self._ref_count = defaultdict(int)
+            else:
+                self._ref_count = None
         else:
-            self._ref_count = None
+            if prune:
+                self._ref_count = ref_count
+            else:
+                raise ValueError("Cannot pass an existing reference count in to a non-pruning trie")
```

Comment:
Maybe a `ValidationError` instead?

Reply:
Depends.

Yes, if this is an exception that you would expect users to run into during normal use of the library (albeit maybe slight mis-use)

No, if this would only occur if 1) someone is mucking with internals or 2) it is only likely to be encountered while working on the core library code.

Reply:
> No, if this would only occur if 1) someone is mucking with internals or 2) it is only likely to be encountered while working on the core library code.

Yeah, I guess it's more of this one. So I'll leave it. :+1: